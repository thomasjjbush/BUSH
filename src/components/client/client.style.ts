import styled from 'styled-components';

export const StyledClient = styled.a<{ background: boolean }>`
    background-color: ${(props): string => (props.background ? 'white' : 'transparent')};
    display: inline-block;
    padding: 10px;
`;
