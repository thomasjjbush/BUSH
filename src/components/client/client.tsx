import React, { FunctionComponent, ReactElement } from 'react';
import { Image } from '../';
import { StyledClient } from './client.style';
import { ClientProps as Props } from '../../types';

export const Client: FunctionComponent<Props> = ({
    alt,
    background = true,
    logo,
    size,
    url,
    width,
}: Props): ReactElement => (
    <StyledClient background={background} {...(url && { href: url })} rel="noreferrer" target="_blank">
        <Image alt={alt} src={logo} width={width} size={size} />
    </StyledClient>
);

Client.displayName = 'Client';
