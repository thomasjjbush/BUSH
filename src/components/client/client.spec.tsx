import React from 'react';
import { shallow } from 'enzyme';
import { Client } from './client';

describe('Client', () => {
    it('should match snapshot', () => {
        expect(
            shallow(<Client alt="alt" background={false} logo="logo" size={20} url="url" width={10} />),
        ).toMatchSnapshot();
    });

    it('should match snapshot with default argument of backrgound', () => {
        expect(shallow(<Client logo="logo" size={20} url="url" width={10} />)).toMatchSnapshot();
    });

    it('should apply css filter if background is false and inverse prop is true', () => {
        expect(
            shallow(<Client alt="alt" background={false} inverse logo="logo" size={20} url="url" width={10} />),
        ).toMatchSnapshot();
    });
});
