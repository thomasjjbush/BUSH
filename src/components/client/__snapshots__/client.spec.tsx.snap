// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client should apply css filter if background is false and inverse prop is true 1`] = `
<styled.a
  background={false}
  href="url"
  inverse={true}
  rel="noreferrer"
  styles="
    background-color: transparent;
    display: inline-block;
    padding: 10px;
    filter: brightness(0) invert(1);
"
  target="_blank"
>
  <Image
    alt="alt"
    size={20}
    src="logo"
    width={10}
  />
</styled.a>
`;

exports[`Client should match snapshot 1`] = `
<styled.a
  background={false}
  href="url"
  rel="noreferrer"
  styles="
    background-color: transparent;
    display: inline-block;
    padding: 10px;
    
"
  target="_blank"
>
  <Image
    alt="alt"
    size={20}
    src="logo"
    width={10}
  />
</styled.a>
`;

exports[`Client should match snapshot with default argument of backrgound 1`] = `
<styled.a
  background={true}
  href="url"
  rel="noreferrer"
  styles="
    background-color: white;
    display: inline-block;
    padding: 10px;
    
"
  target="_blank"
>
  <Image
    size={20}
    src="logo"
    width={10}
  />
</styled.a>
`;
